### PR TITLE
feat: Update xCall and IBC connection to lates spec and update spec

### PR DESCRIPTION
**Pull request by AntonAndell**
Add fee logic to connection, allow xCall to be used the same as original xCall by introducing optional parameters and admin controlled default protocols. Use per connection timeout height based on each lightclient.

### Commit Message

```bash
type: commit message
```

see the [guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#commit-messages) for commit messages.

### Changelog Entry

```bash
version: <log entry>
```

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have documented my code in accordance with the [documentation guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#documentation)
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run the unit tests
- [ ] I only have one commit (if not, squash them into one commit).
- [ ] I have a descriptive commit message that adheres to the [commit message guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#commit-messages)

> Please review the [CONTRIBUTING.md](/CONTRIBUTING.md) file for detailed contributing guidelines.

### CODE REVIEW

File: docs/adr/XCall_IBC_Connection.md [hunk 1]
```diff
@@ -337,20 +379,32 @@ public void transferAdmin(Address admin) {
...
+ * Configures a ibc connection so that a channel can be establised for a specigic nid
+
+
+ * @param connectionId The connection id of the connection/chain to open the connection to
+ * @param portId  The allocated port name of the connection on the counterparty chain
+ * @param counterpartyNid The network Id to be associated with this connection
+ * @param clientId The lightClient associated with this connection
+ * @param timeoutHeight The timeoutheight to be used on packets, it is recommended to use a high value similar to the trusting period of the lightclient.
+ */
+public void configureConnection(String connectionId, String portId, String counterpartyNid, String clientId, BigInteger timeoutHeight) {
```
Comment (redlarva) [inline docs/adr/XCall_IBC_Connection.md:392]: Can we change the name of portId to counterpartyPortId?

File: contracts/javascore/xcall-connection/src/main/java/ibc/xcall/connection/IBCConnection.java [hunk 2]
```diff
@@ -72,132 +81,196 @@ private void onlyAdmin() {
...
+
         Packet packet = Packet.decode(calldata);
         Message msg = Message.fromBytes(packet.getData());
-        String nid = counterPartyNetworkId.get(packet.getDestinationChannel());
+        String nid = networkIds.get(packet.getDestinationChannel());
         Context.require(nid != null);
-        if (!msg.getSn().equals(BigInteger.ZERO)) {
+
+        BigInteger unclaimedFees = unclaimedPacketFees.at(nid).getOrDefault(relayer, BigInteger.ZERO);
+        unclaimedPacketFees.at(nid).set(relayer, unclaimedFees.add(msg.getFee()));
```
[the remark this answers is not in the document]
Comment (AntonAndell) [reply contracts/javascore/xcall-connection/src/main/java/ibc/xcall/connection/IBCConnection.java:158]: Good catch, but it should only be added is sn is not null. So logic should be moved to after theif (msg.getSn() == null) clause i think

File: docs/adr/XCall_IBC_Connection.md (hunk 1 above)
Comment (AntonAndell) [reply docs/adr/XCall_IBC_Connection.md:392]: Yeah that would be more clear

**Review by redlarva — APPROVED** [review]
looks good to me

**Comment by AntonAndell** [comment]
You can hold of on merge found some small issues

**Review by ibrizsabin — APPROVED** [review]

**Pull request merged**